Sync/Synchronizers/Bookmarks/BookmarksSynchronizer.swift

```diff
@@ -89,6 +89,51 @@ open class BufferingBookmarksSynchronizer: TimestampedSingleCollectionSynchroniz
         return BookmarksStorageVersion
     }
 
+    fileprivate func createMobileRootRecord(_ storage: LocalItemSource, _ buffer: BufferItemSource, newBookmarks: [BookmarkMirrorItem]) -> Deferred<Maybe<Record<BookmarkBasePayload>>> {
+        let newBookmarkGUIDS = newBookmarks.map { $0.guid }
+        return buffer.getBufferItemWithGUID(BookmarkRoots.MobileFolderGUID).bind { maybeMobileRoot in
+            // Update (or create!) the Mobile Root folder with its new children.
+            if let mobileRoot = maybeMobileRoot.successValue {
+                return buffer.getBufferChildrenGUIDsForParent(mobileRoot.guid) >>== { curChildren in
+                    return deferMaybe((mobileRoot: mobileRoot, children: curChildren + newBookmarkGUIDS))
```

Comment:
A closure with only a single statement can omit the `return`. And consider using `.map` (untested):

```
return buffer.getBufferChildrenGUIDsForParent(mobileRoot.guid)
             .map { $0.map({ (mobileRoot: mobileRoot, children: $0 + newBookmarkGUIDs)) } }
```